rs/embedders/benches/heap.rs

```diff
@@ -146,34 +189,44 @@ fn bench(c: &mut C, mem: Mem, call: Call, op: Op, dir: Dir, size: Size, step: St
     let throughput = throughput(size, step);
     let setup_action = setup_action(src);
 
-    let op = heap_op(op);
     let func_init_body = heap_func_init_body(mem, size, src);
-    let loop_body = loop_body(&op, mem, dir, size, step);
+    let op = heap_op(op, mem);
+    let loop_body = loop_body(&op, dir, 0, size, step);
     let canister_init_body = heap_canister_init_body(mem, size, src);
+    let canister_setup_body = heap_canister_setup_body(mem, size, src);
     let memory_body = heap_memory_body(mem, size, src);
     let wat = format!(
         r#"
         (module
-            (import "ic0" "msg_reply" (func $ic0_msg_reply))
+            (import "ic0" "msg_reply" (func $msg_reply))
             (global $counter (mut i64) (i64.const 42))
+            (global $data (mut i64) (i64.const 0))
             (func (export "canister_update update_empty")
-                (call $ic0_msg_reply)
+                (call $msg_reply)
+            )
+            (func (export "canister_init")
+                (local $address i64)
+                (global.set $counter (i64.add (global.get $counter) (i64.const 1)))
```

Comment:
Why is the counter bumped here if its initial value is 42 (arbitrary) anyway?

Reply:
It’s just a pattern to make sure we never rewrite the same data.